File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        return self.search_insert_recursive(nums, target, 0, len(nums)-1)
    
    def search_insert_recursive(self, nums: List[int], target: int, left: int, right: int) -> int:
        if right <= left:
            if nums[left] < target:
                return left+1
            return left
        
        midpoint = (left+right)//2
        if nums[midpoint] == target:
            return midpoint
        if nums[midpoint] < target:
            return self.search_insert_recursive(nums, target, midpoint+1, right)
        return self.search_insert_recursive(nums, target, left, midpoint-1)

File: 0035-search-insert-position/0035-search-insert-position.go
func searchInsert(nums []int, target int) int {
    return searchInsertRecursive(nums, target, 0, len(nums)-1)
}

func searchInsertRecursive(nums []int, target, low, high int) int {
    if high <= low {
        if nums[low] < target {
            return low+1
        }
        return low
    }
    midpoint := (low+high)/2
    if nums[midpoint] == target {
        return midpoint
    }
    if nums[midpoint] < target {
        return searchInsertRecursive(nums, target, midpoint+1, high)
    }
    return searchInsertRecursive(nums, target, 0, midpoint-1)
}